docs/user/alerting/alerting-troubleshooting.asciidoc

```diff
@@ -69,3 +69,168 @@ Configuration options are available to specialize connections to TLS servers,
 including ignoring server certificate validation, and providing certificate
 authority data to verify servers using custom certificates.  For more details, 
 see <<action-settings,Action settings>>.
+
+[float]
+[[rules-long-execution-time]]
+=== Identify long running rules
+
+The following query can help you identify rules that are taking a long time to execute and may be impacting the overall health of your deployment.
+
+Query for a list of rule ids, bucketed by their execution times:
+
+[source,sh]
+--------------------------------------------------
+$ GET /.kibana-event-log*/_search
+{
+  "size": 0,
+  "query": {
+    "bool": {
+      "filter": [
+        {
+          "range": {
+            "@timestamp": {
+              "gte": "now-1d", <1>
+              "lte": "now"
+            }
+          }
+        },
+        {
+          "term": {
+            "event.action": {
+              "value": "execute"
+            }
+          }
+        }
+      ]
+    }
+  },
+  "aggs": {
+    "ruleIdsByExecutionDuration": {
+      "histogram": {
+        "field": "event.duration",
+        "interval": 10000000000 <2>
+      },
+      "aggs": {
+        "ruleId": {
+          "nested": {
+            "path": "kibana.saved_objects"
+          },
+          "aggs": {
+            "ruleId": {
+              "terms": {
+                "field": "kibana.saved_objects.id",
+                "size": 10 <3>
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
```

Comment:
The first time I ran this, all the executions fit in the first bucket, so ... not much value there.  Turned the interval down to get more buckets created.

Could we used a fixed bucket count here?  Like 10?  So you'd always see 10 buckets, with ES figuring out the interval.  I can't remember if that's a thing you can do in ES, but I think so?

Reply:
I believe date histogram allows this type of auto-interval calculation but the regular histogram aggregation does not. There is an issue for it in Elasticsearch https://github.com/elastic/elasticsearch/issues/31828